lib/modules/webserver/index.js

```diff
@@ -17,14 +17,27 @@ class WebServer {
     this.port = options.port || this.webServerConfig.port;
 
     this.events.emit("status", __("Starting Server"));
-    this.server = new Server({host: this.host, port: this.port});
 
-    this.setServiceCheck();
-    this.listenToCommands();
-    this.registerConsoleCommands();
+    this.testPort(() => {
+      this.server = new Server({host: this.host, port: this.port});
 
-    let self = this;
-    this.server.start((_err, message) => self.logger.info(message));
+      this.setServiceCheck();
+      this.listenToCommands();
+      this.registerConsoleCommands();
+      this.server.start((_err, message) => this.logger.info(message));
+    });
+  }
+
+  testPort(done) {
+    utils.pingEndpoint(this.host, this.port, 'http', 'http', '', (err) => {
+      if (err) { // Port is ok
+        return done();
+      }
+      this.logger.warn(__('Webserver already running on port {{oldPort}}. Trying to connect on {{newPort}}',
+        {oldPort: this.port, newPort: this.port + 1}));
```

Comment:
I'm wondering if it would make sense to use `server.listen(0)` so the system returns an available port to us instead.

Reply:
Oh, didn't know it could do that. I'll try

Reply:
It worked. Good idea